Source/ClientLib/CMakeLists.txt

```diff
@@ -17,7 +17,11 @@ target_include_directories(ClientLib
 )
 
 # Inherit Shared's precompiled header.
-target_precompile_headers(ClientLib REUSE_FROM SharedLib)
+# CMake causes issues when using precompiled headers with GCC on macOS,
+# so precompiled headers as disabled for that target.
+if (NOT APPLE OR CMAKE_C_COMPILER_ID MATCHES "Clang|AppleClang")

```

Comment:
Is `CMAKE_CXX_COMPILER_ID` equivalent here? If so, I'd prefer to use it instead of `CMAKE_C_COMPILER_ID`.

Reply:
Although they're the same in practice, I agree that CMAKE_CXX_COMPILER_ID would be better.